src/S3/Exception/S3MultipartUploadException.php

```diff
@@ -0,0 +1,86 @@
+<?php
+namespace Aws\S3\Exception;
+
+use Aws\CommandInterface;
+use Aws\Exception\AwsException;
+use Aws\Multipart\UploadState;
+
+class S3MultipartUploadException extends \Aws\Exception\MultipartUploadException
+{
+    /** @var string Bucket of the transfer object */
+    private $bucket;
+    /** @var string Key of the transfer object */
+    private $key;
+    /** @var string Source file name of the transfer object */
+    private $filename;
+
+    /**
+     * @param UploadState      $state Upload state at time of the exception.
+     * @param \Exception|array $prev  Exception being thrown. Could be an array of
+     *                                AwsExceptions being thrown when uploading parts
+     *                                for one object, or an instance of AwsException
+     *                                for a specific Multipart error being thrown in
+     *                                the MultipartUpload process.
+     */
+    public function __construct(UploadState $state, $prev = null) {
+        if (is_array($prev)) {
+            foreach ($prev as $part => $error) {
+                $this->collectPathInfo($error->getCommand());
```

Comment:
Make sense, I used to make it exactly like `MultipartException` scenario, taking a second look, it does provides enough info. in this way. I'll fix that : )